test/verify/check-machines-dbus

```diff
@@ -188,6 +188,73 @@ class TestMachinesDBus(machineslib.TestMachines):
         b.wait_not_in_text("#vm-subVmTest1-boot-order", bootOrder)
         b.wait_not_present("#boot-order-tooltip")
 
+    def testDomainMemorySettings(self):
+        b = self.browser
+
+        args = self.startVm("subVmTest1")
+
+        self.login_and_go("/machines")
+        b.wait_in_text("body", "Virtual Machines")
+        b.wait_present("tbody tr th")
+        b.wait_in_text("tbody tr th", "subVmTest1")
+
+        b.click("tbody tr th") # click on the row header
+        b.wait_present("#vm-subVmTest1-state")
+        b.wait_in_text("#vm-subVmTest1-state", "running")
+
+        b.wait_present("#vm-subVmTest1-overview") # wait for the tab
+        b.click("#vm-subVmTest1-overview") # open the "Overview" subtab
+
+        # Wait for the edit link
+        b.wait_present("#vm-subVmTest1-memory-count")
+        b.click("#vm-subVmTest1-memory-count")
+
+        # Change memory
+        b.wait_present("#vm-subVmTest1-memory-modal-memory")
+        # VM initially should have 256MiB
+        current_memory = int(b.attr("#vm-subVmTest1-memory-modal-memory", "value"))
+        b.wait_attr("#vm-subVmTest1-memory-modal-max-memory", "readonly", "")
+
+        # Check memory hotunplugging
+        # The balloon driver needs to be loaded to descrease memory
+        wait(lambda: "Linux version" in self.machine.execute("cat {0}".format(args["logfile"])), delay=3)
+
+        b.set_input_text("#vm-subVmTest1-memory-modal-memory", str(current_memory - 10))
+        # Save the memory settings
+        b.click("#vm-subVmTest1-memory-modal-save")
+        b.wait_not_present("#vm-memory-modal")
+
+        b.wait_present("#vm-subVmTest1-memory-count")
+        b.wait_in_text("#vm-subVmTest1-memory-count", "{0} MiB".format(current_memory - 10))
+
+        # Shut off domain and check changes are  still there
+        b.click("#vm-subVmTest1-off-caret")
+        b.wait_visible("#vm-subVmTest1-forceOff")
+        b.click("#vm-subVmTest1-forceOff")
+        b.wait_in_text("#vm-subVmTest1-state", "shut off")
+        b.wait_in_text("#vm-subVmTest1-memory-count", "{0} MiB".format(current_memory - 10))
+
+        # Click for the edit link
+        b.click("#vm-subVmTest1-memory-count")
+
+        # Verify that limiting max memory in offline VMs bellow memory will decrease memory as well
+        b.set_input_text("#vm-subVmTest1-memory-modal-max-memory", str(current_memory - 20))
+        self.assertEqual(int(b.attr("#vm-subVmTest1-memory-modal-memory", "value")), current_memory - 20)
```

Comment:
This looks like a race condition. Even if the code somehow makes sure that memory is  synchronously updated with max-memory, we shouldn't rely on that. Please use `b.wait_attr()` here and below.

Reply:
Done